crates/matrix-sdk-ui/src/timeline/beacons.rs

```diff
@@ -0,0 +1,97 @@
+//! This module handles rendering of MSC3489 live location sharing in the
+//! timeline.
```

Comment:
I don't think having this in the timeline is the proper cutting point. Instead, this would be more useful to have at the SDK layer, and maybe even with a slightly different API.